### PR TITLE
feat: add elevation profile function of distance and not time

### PR DESCRIPTION
**Pull request by JulesL2**
Hi,
I have add the support for drawing elevation profile in distance rather than in points.

It resolves the issue https://github.com/ThomasParistech/pretty-gpx/issues/3

### CODE REVIEW

File: pretty_gpx/drawing/poster_image_cache.py [hunk 1]
```diff
@@ -174,9 +174,9 @@ def from_gpx_data(gpx_data: AugmentedGpxData,
         ele_scatter, ele_fill_poly, stats = get_elevation_drawings(layout=layout,
                                                                    h_pix=h, w_pix=w,
                                                                    list_ele=gpx_data.track.list_ele,
+                                                                   list_cum_d=gpx_data.track.list_cumul_d,
                                                                    passes_ids=gpx_data.passes_ids,
                                                                    huts_ids=gpx_data.hut_ids,
-                                                                   daily_dist_km=gpx_data.daily_dist_km,
```
Comment (ThomasParistech) [inline pretty_gpx/drawing/poster_image_cache.py:179]: The field `daily_dist_km` from AugmentedGpxData is no longer used. You can remove it

File: pretty_gpx/gpx/augmented_gpx_data.py [hunk 2]
```diff
@@ -250,18 +250,15 @@ def find_huts_between_daily_tracks(list_gpx_path: list[str] | list[bytes],
...
-                              list_lat=[lat
-                                        for gpx in list_gpx_track
-                                        for lat in gpx.list_lat],
-                              list_ele=[ele
-                                        for gpx in list_gpx_track
-                                        for ele in gpx.list_ele])
-
+    assert(abs(1-(full_gpx_track.list_cumul_ele[-1]-full_uphill_m)/
+                full_gpx_track.list_cumul_ele[-1]) < 0.01, 
+                "Error on the total climb")
```
Comment (ThomasParistech) [inline pretty_gpx/gpx/augmented_gpx_data.py:261]: Remove `dist_km, uphill_m` returned by `GpxTrack.load` and add properties to access it via the cumulative lists. Then no need to assert :)

```
    @property
    def dist_km(self) -> float:
        """Total distance in km."""
        return self.list_cumul_dist[-1]
```

File: pretty_gpx/gpx/gpx_track.py [hunk 3]
```diff
@@ -46,6 +51,32 @@ def load(gpx_path: str | bytes) -> tuple['GpxTrack', float, float]:
...
+                    gpx_track.list_cumul_d.append(previous_cum_d+d)
+                    gpx_track.list_cumul_ele.append(previous_cum_ele+h_diff)
+
+                    previous_cum_d += d
+                    previous_cum_ele += h_diff
+
+                    point_prev = point
+                all_segment_cumul_d += previous_cum_d
+                all_segment_cumul_ele += previous_cum_ele
+
```
Comment (ThomasParistech) [inline pretty_gpx/gpx/gpx_track.py:79]: maybe you can add the assert at the end of this method to compare the cum sums against `gpx.length_3d()*1e-3` and `gpx.get_uphill_downhill().uphill` ?

File: pretty_gpx/gpx/gpx_track.py [hunk 4]
```diff
@@ -46,6 +51,32 @@ def load(gpx_path: str | bytes) -> tuple['GpxTrack', float, float]:
...
+                                                point.elevation)*1e-3
+                        h_diff = point.elevation - point_prev.elevation
+                        if h_diff < 0:
+                            h_diff = 0.0
+                    
+                    gpx_track.list_cumul_d.append(previous_cum_d+d)
+                    gpx_track.list_cumul_ele.append(previous_cum_ele+h_diff)
+
+                    previous_cum_d += d
+                    previous_cum_ele += h_diff
```
Comment (ThomasParistech) [inline pretty_gpx/gpx/gpx_track.py:74]: append after applying the `+=` operator

**Review by ThomasParistech — CHANGES_REQUESTED** [review]
Many thanks!

File: pretty_gpx/gpx/augmented_gpx_data.py [hunk 5]
```diff
@@ -226,13 +223,11 @@ def find_huts_between_daily_tracks(list_gpx_path: list[str] | list[bytes],
     list_dist_km: list[float] = []
     list_uphill_m: list[float] = []
     for path in list_gpx_path:
-        gpx_track, dist_km, uphill_m = GpxTrack.load(path)
+        gpx_track = GpxTrack.load(path)
         list_gpx_track.append(gpx_track)
```
Comment (ThomasParistech) [inline pretty_gpx/gpx/augmented_gpx_data.py:227]: a list comprehension would be better now that we remove the other two :)

File: pretty_gpx/gpx/gpx_track.py [hunk 6]
```diff
@@ -87,7 +87,15 @@ def load(gpx_path: str | bytes) -> tuple['GpxTrack', float, float]:
             plt.ylabel('Elevation (in m)')
             plt.show()
 
-        return gpx_track, gpx.length_3d()*1e-3, gpx.get_uphill_downhill().uphill
+        print(all_segment_cumul_d,gpx.length_3d()*1e-3)
+        print(all_segment_cumul_ele,gpx.get_uphill_downhill().uphill)
```
Comment (ThomasParistech) [inline pretty_gpx/gpx/gpx_track.py:91]: make the log more readable :art:

**Review by ThomasParistech — APPROVED** [review]
:ok_hand:

File: pretty_gpx/gpx/augmented_gpx_data.py (hunk 5 above)
Comment (JulesL2) [reply pretty_gpx/gpx/augmented_gpx_data.py]: fixed in the last commit

File: pretty_gpx/gpx/gpx_track.py (hunk 6 above)
Comment (JulesL2) [reply pretty_gpx/gpx/gpx_track.py]: Firstly I tought to remove it but now I've added text, is there a common DEBUG variable ?

**Pull request merged**